tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -235,7 +247,18 @@ declare var cfg:ISnippetConfig;
                 return scriptElement;
             }
 
-            let theScript = _createScript();
+            let theScript = _createScript(targetSrc);
+
+            if (appInsights.cr && loadFailed && domains.some(endpoint => targetSrc.indexOf(endpoint) !== -1)){
+                for (const currentEndpoint of domains) {
+                    loadFailed = false;
+                    theScript = _createScript("https://" + currentEndpoint + "/scripts/b/ai.2.min.js");
+                    if (!loadFailed) {
```

Comment:
I'm not sure what is going on here, the `_createScript` will return synchronously, so checking the `loadFailed` here is not going to work properly.

I'm also not following how this would "create" a new script for the next CDN entry, only AFTER the previous one failed. I was expecting that we would need to add some logic to the `_handleError` function so that instead of reporting a failure it would trigger a new script tag to be created at that point and added to the page and only after running out of domains to "try" would it then report the failure.